In [1]:
# -*-coding:utf-8 -*

#Création de l'environnement de travail
import numpy as np
import matplotlib as plt
import pandas as pd
# Importation des librairies



In [2]:
df = pd.read_csv("base.csv", header = 0) #Lecture du fichier csv avec prise en compte de l'en-tête comme index

In [3]:
del df["Code Domaine"]
del df["Domaine"]


In [4]:
a = df.pivot(columns = 'Élément', values = 'Valeur')
df = df.join(a)
del df["Unité"]
del df["Valeur"]
del df["Élément"]
del df["Code Année"]
del df["Code Élément"]

In [5]:
#df['population'] = df['Population totale']*1000
#del df["Population totale"]



In [6]:
df['Nature Produit'] = ((df['Code Produit'] == 2511) | (df['Code Produit'] == 2805) | (df['Code Produit'] == 2513) | (df['Code Produit'] == 2514) | (df['Code Produit'] == 2515) | (df['Code Produit'] == 2516) | (df['Code Produit'] == 2517) | (df['Code Produit'] == 2518) | (df['Code Produit'] == 2520))
df = df.replace({'Nature Produit': {True: 'céréales'}})
df = df.replace({'Nature Produit': {False: 'autre'}})

In [7]:
df = df.groupby(['Code Pays', 'Pays', 'Année', 'Produit', 'Code Produit','Nature Produit']).sum().reset_index()


In [8]:
a = df[['Code Pays','Pays','Produit','Population totale']]
df_population = a[a['Produit'] == 'Population']
del df_population['Produit']
df_population.reset_index(drop = True, inplace = True)
df_population

Code Pays                                    Pays  Population totale
0            1                                 Arménie             2977.0
1            2                             Afghanistan            30552.0
2            3                                 Albanie             3173.0
3            4                                 Algérie            39208.0
4            7                                  Angola            21472.0
5            8                      Antigua-et-Barbuda               90.0
6            9                               Argentine            41446.0
7           10                               Australie            23343.0
8           11                                Autriche             8495.0
9           12                                 Bahamas              377.0
10          14                                 Barbade              285.0
11          16                              Bangladesh           156595.0
12          17                                Bermudes               65.0
13          19         Bolivie (État plurinational de)            10671.0
14          20                                Botswana             2021.0
15          21                                  Brésil           200362.0
16          23                                  Belize              332.0
17          25                            Îles Salomon              561.0
18          26                       Brunéi Darussalam              418.0
19          27                                Bulgarie             7223.0
20          28                                 Myanmar            53259.0
21          32                                Cameroun            22254.0
22          33                                  Canada            35182.0
23          35                              Cabo Verde              499.0
24          37               République centrafricaine             4616.0
25          38                               Sri Lanka            21273.0
26          39                                   Tchad            12825.0
27          40                                   Chili            17620.0
28          44                                Colombie            48321.0
29          46                                   Congo             4448.0
..         ...                                     ...                ...
141        210                                   Suède             9571.0
142        211                                  Suisse             8078.0
143        213                            Turkménistan             5240.0
144        215             République-Unie de Tanzanie            49253.0
145        216                               Thaïlande            67011.0
146        217                                    Togo             6817.0
147        220                       Trinité-et-Tobago             1341.0
148        221                                    Oman             3632.0
149        222                                 Tunisie            10997.0
150        223                                 Turquie            74933.0
151        225                     Émirats arabes unis             9346.0
152        226                                 Ouganda            37579.0
153        229                             Royaume-Uni            63384.0
154        230                                 Ukraine            45239.0
155        231                   États-Unis d'Amérique           320051.0
156        233                            Burkina Faso            16935.0
157        234                                 Uruguay             3407.0
158        235                             Ouzbékistan            28934.0
159        236  Venezuela (République bolivarienne du)            30405.0
160        237                                Viet Nam            91680.0
161        238                                Éthiopie            94101.0
162        244                                   Samoa              190.0
163      

In [9]:
df = df[df['Produit'] != 'Population']
del df['Population totale']

In [10]:
df = df.merge(df_population)

In [11]:
df = df.rename(columns = {'Aliments pour animaux' : 'Aliments pour animaux (en milliers de tonnes)', 'Autres Utilisations': 'Autres Utilisations (en milliers de tonnes)', 'Disponibilité intérieure' : 'Disponibilité intérieure (en milliers de tonnes)', 'Nourriture' : 'Nourriture (en milliers de tonnes)', 'Pertes' : 'Pertes (en milliers de tonnes)', 'Semences' : 'Semences (en milliers de tonnes)', 'Traitement' : 'Traitement (en milliers de tonnes)', 'Population totale' : 'Population totale (en milliers)'})

In [12]:
df

Code Pays     Pays  Année                   Produit  Code Produit  \
0              1  Arménie   2013          Abats Comestible          2736   
1              1  Arménie   2013           Agrumes, Autres          2614   
2              1  Arménie   2013    Alcool, non Comestible          2659   
3              1  Arménie   2013     Aliments pour enfants          2680   
4              1  Arménie   2013                    Ananas          2618   
5              1  Arménie   2013  Animaux Aquatiques Autre          2769   
6              1  Arménie   2013    Arachides Decortiquees          2556   
7              1  Arménie   2013                    Avoine          2516   
8              1  Arménie   2013                   Bananes          2615   
9              1  Arménie   2013         Bananes plantains          2616   
10             1  Arménie   2013              Beurre, Ghee          2740   
11             1  Arménie   2013                     Bière          2656   
12             1  Arménie   2013                       Blé          2511   
13             1  Arménie   2013      Boissons Alcooliques          2658   
14             1  Arménie   2013        Boissons Fermentés          2657   
15             1  Arménie   2013                      Café          2630   
16             1  Arménie   2013              Cephalopodes          2766   
17             1  Arménie   2013           Citrons & Limes          2612   
18             1  Arménie   2013        Coco (Incl Coprah)          2560   
19             1  Arménie   2013                 Crustacés          2765   
20             1  Arménie   2013                     Crème          2743   
21             1  Arménie   2013          Céréales, Autres          2520   
22             1  Arménie   2013                    Dattes          2619   
23             1  Arménie   2013        Edulcorants Autres          2543   
24             1  Arménie   2013             Feve de Cacao          2633   
25             1  Arménie   2013            Fruits, Autres          2625   
26             1  Arménie   2013                  Girofles          2642   
27             1  Arménie   2013    Graines Colza/Moutarde          2558   
28             1  Arménie   2013      Graines de tournesol          2557   
29             1  Arménie   2013    Graisses Animales Crue          2737   
...          ...      ...    ...                       ...           ...   
15295        351    Chine   2013                   Piments          2641   
15296        351    Chine   2013        Plantes Aquatiques          2775   
15297        351    Chine   2013  Plantes Oleiferes, Autre          2570   
15298        351    Chine   2013                      Pois          2547   
15299        351    Chine   2013        Poissons Eau Douce          2761   
15300        351    Chine   2013   Poissons Marins, Autres          2764   
15301        351    Chine   2013       Poissons Pelagiques          2763   
15302        351    Chine   2013                    Poivre          2640   
15303        351    Chine   2013                    Pommes          2617   
15304        351    Chine   2013           Pommes de Terre          2531   
15305        351    Chine   2013              Racines, Nda          2534   
15306        351    Chine   2013                    Raisin          2620   
15307        351    Chine   2013          Riz (Eq Blanchi)          2805   
15308        351    Chine   2013                    Seigle          2515   
15309        351    Chine   2013                      Soja          2555   
15310        351    Chine   2013                    Sorgho          2518   
15311        351    Chine   2013             Sucre Eq Brut          2542   
15312        351    Chine   2013      Sucre non centrifugé          2541   
15313        351    Chine   2013          Sucre, betterave          2537   
15314        351    Chine   2013              Sucre, canne          2536   
15315        351    Chine   2013                    Sés

## Population mondiale en 2013

In [13]:
population_mondiale_2013_en_milliers = df_population['Population totale'].sum()
population_mondiale_2013_en_milliers

6997326.0

Ne pas prendre en compte les différentes zones de la chine

Calculez (pour chaque pays et chaque produit) la disponibilité alimentaire en kcal. Vous le ferez à partir de ces informations :

    - Population de chaque pays ;

    - Disponibilité alimentaire ("Food Supply" en anglais) donnée pour chaque produit et pour chaque pays en kcal/personne/jour ,
    


A faire : Identifiez ces redondances, en donnant par exemple votre réponse sous forme de formule mathématique

Parmi les documents sur les Bilans alimentaires que vous avez téléchargés,il y a des informations redondantes. En effet, certaines peuvent se calculer à partir d'autres.

Disponibilité intérieure = Production + Importations - Exportations + Evolution des stocks

Nourriture = Disponibilité intérieure - Semences - Pertes - Aliments pour animaux - Autre Utilisation - Traitement

Disponibilité alimentaire = Nourriture / Population

In [14]:
df['Disponibilité alimentaire (Kcal/an)'] = df['Disponibilité alimentaire (Kcal/personne/jour)'] *365 * df['Population totale (en milliers)']
df['Disponibilité alimentaire en quantité (kg/an)'] = df['Disponibilité alimentaire en quantité (kg/personne/an)']* df['Population totale (en milliers)']

A partir de cette dernière information, et à partir du poids de la disponibilité alimentaire (pour chaque pays et chaque produit), calculez pour chaque produit le ratio "énergie/poids", que vous donnerez en kcal/kg.

In [15]:
df['ratio_energie_poids en Kcal/an'] = df['Disponibilité alimentaire (Kcal/an)'] / df['Disponibilité alimentaire en quantité (kg/an)']

A faire : Donnez le ratio énergie/poids pour 5 produits de votre choix.

In [16]:
a = df[['Produit','Disponibilité alimentaire en quantité (kg/an)','Disponibilité alimentaire (Kcal/an)']]
a = a.groupby(['Produit']).sum()
a['ratio_energie_poids en Kcal/an'] = a['Disponibilité alimentaire (Kcal/an)'] / a['Disponibilité alimentaire en quantité (kg/an)']
a.head(6)

Disponibilité alimentaire en quantité (kg/an)  \
Produit                                                                   
Abats Comestible                                            15669030.53   
Agrumes, Autres                                             11364922.93   
Alcool, non Comestible                                             0.00   
Aliments pour enfants                                         761644.43   
Ananas                                                      20779937.54   
Animaux Aquatiques Autre                                     1251630.33   

                          Disponibilité alimentaire (Kcal/an)  \
Produit                                                         
Abats Comestible                                 1.756899e+10   
Agrumes, Autres                                  2.479442e+09   
Alcool, non Comestible                           0.000000e+00   
Aliments pour enfants                            2.763215e+09   
Ananas                                           6.963713e+09   
Animaux Aquatiques Autre                         5.599158e+08   

                          ratio_energie_poids en Kcal/an  
Produit                                                   
Abats Comestible                             1121.255429  
Agrumes, Autres                               218.166235  
Alcool, non Comestible                               NaN  
Aliments pour enfants                        3627.958804  
Ananas                                        335.117150  
Animaux Aquatiques Autre                      447.349211

In [17]:
#Vérification Nourriture = Dispo Alim Kg an
b = df[['Produit','Nourriture (en milliers de tonnes)','Disponibilité alimentaire en quantité (kg/an)','Disponibilité alimentaire (Kcal/an)']]
b = b.groupby(['Produit']).sum()
b.head()

Nourriture (en milliers de tonnes)  \
Produit                                                      
Abats Comestible                                   15669.0   
Agrumes, Autres                                    11360.0   
Alcool, non Comestible                                 0.0   
Aliments pour enfants                                762.0   
Ananas                                             20757.0   

                        Disponibilité alimentaire en quantité (kg/an)  \
Produit                                                                 
Abats Comestible                                          15669030.53   
Agrumes, Autres                                           11364922.93   
Alcool, non Comestible                                           0.00   
Aliments pour enfants                                       761644.43   
Ananas                                                    20779937.54   

                        Disponibilité alimentaire (Kcal/an)  
Produit                                                      
Abats Comestible                               1.756899e+10  
Agrumes, Autres                                2.479442e+09  
Alcool, non Comestible                         0.000000e+00  
Aliments pour enfants                          2.763215e+09  
Ananas                                         6.963713e+09

Étonnamment, il arrive que ce ratio soit différent en fonction du pays. Ce n'est pas très cohérent, mais tant pis. Vous êtes libre de donner le résultat que vous souhaitez (donner les valeurs pour un pays donné, ou faire une moyenne sur tous les pays après avoir retiré les valeurs égales à 0, etc.).

In [18]:
a = a = df[['Pays','Produit','Disponibilité alimentaire en quantité (kg/an)','Disponibilité alimentaire (Kcal/an)','ratio_energie_poids en Kcal/an']]
a = a[a['Pays'] == 'France']
a

Pays                   Produit  \
4048  France          Abats Comestible   
4049  France           Agrumes, Autres   
4050  France    Alcool, non Comestible   
4051  France     Aliments pour enfants   
4052  France                    Ananas   
4053  France  Animaux Aquatiques Autre   
4054  France    Arachides Decortiquees   
4055  France                    Avoine   
4056  France                   Bananes   
4057  France         Bananes plantains   
4058  France              Beurre, Ghee   
4059  France                     Bière   
4060  France                       Blé   
4061  France      Boissons Alcooliques   
4062  France        Boissons Fermentés   
4063  France                      Café   
4064  France              Cephalopodes   
4065  France           Citrons & Limes   
4066  France        Coco (Incl Coprah)   
4067  France                 Crustacés   
4068  France                     Crème   
4069  France          Céréales, Autres   
4070  France                    Dattes   
4071  France        Edulcorants Autres   
4072  France             Feve de Cacao   
4073  France            Fruits, Autres   
4074  France                  Girofles   
4075  France    Graines Colza/Moutarde   
4076  France          Graines de coton   
4077  France      Graines de tournesol   
...      ...                       ...   
4113  France                 Perciform   
4114  France                   Piments   
4115  France        Plantes Aquatiques   
4116  France  Plantes Oleiferes, Autre   
4117  France                      Pois   
4118  France        Poissons Eau Douce   
4119  France   Poissons Marins, Autres   
4120  France       Poissons Pelagiques   
4121  France                    Poivre   
4122  France                    Pommes   
4123  France           Pommes de Terre   
4124  France              Racines, Nda   
4125  France                    Raisin   
4126  France          Riz (Eq Blanchi)   
4127  France                    Seigle   
4128  France                      Soja   
4129  France                    Sorgho   
4130  France             Sucre Eq Brut   
4131  France          Sucre, betterave   
4132  France              Sucre, canne   
4133  France                    Sésame   
4134  France                       Thé   
4135  France                   Tomates   
4136  France    Viande d'Ovins/Caprins   
4137  France          Viande de Bovins   
4138  France          Viande de Suides   
4139  France       Viande de Volailles   
4140  France             Viande, Autre   
4141  France                       Vin   
4142  France            Épices, Autres   

      Disponibilité alimentaire en quantité (kg/an)  \
4048                                      376745.26   
4049                                        7072.01   
4050                                           0.00   
4051                                           0.00   
4052                                      167156.60   
4053                                        4500.37   
4054                                       68148.46   
4055                                       36645.87   
4056                                      298310.24   
4057                                       38574.60   
4058                                      513042.18   
4059                                     1500551.94   
4060                                     6971073.13   
4061                                      145297.66   
4062                                      500183.98   
4063                                      358743.78   
4064                                       30859.68   
4065                                      136296.92   
4066                                       25716.40   
4067                                      235305.06   
4068                                      257164.00   
4069                                       46932.43   
4070                                       16715.66   
4071                                      126010.36   
4072                        

A partir de l'information sur la disponibilité alimentaire en kcal (par pays et par produit) que vous avez déterminée, calculez, pour les produits végétaux uniquement, la production mondiale exprimée en kcal.

In [19]:
df4 = pd.read_csv("vegetaux.csv", header = 0) #Lecture du fichier csv avec prise en compte de l'en-tête comme index
df4 = df4.groupby(['Code Produit', 'Produit']).sum().reset_index()
df4 = df4[['Code Produit','Produit']]
df4['Nature Produit 2'] = 'végétal'
df4

Code Produit                  Produit Nature Produit 2
0           2511                      Blé          végétal
1           2513                     Orge          végétal
2           2514                     Maïs          végétal
3           2515                   Seigle          végétal
4           2516                   Avoine          végétal
5           2517                   Millet          végétal
6           2518                   Sorgho          végétal
7           2520         Céréales, Autres          végétal
8           2531          Pommes de Terre          végétal
9           2532                   Manioc          végétal
10          2533           Patates douces          végétal
11          2534             Racines, Nda          végétal
12          2535                  Ignames          végétal
13          2536             Sucre, canne          végétal
14          2537         Sucre, betterave          végétal
15          2541     Sucre non centrifugé          végétal
16          2542            Sucre Eq Brut          végétal
17          2543       Edulcorants Autres          végétal
18          2546                 Haricots          végétal
19          2547                     Pois          végétal
20          2549      Légumineuses Autres          végétal
21          2551                     Noix          végétal
22          2555                     Soja          végétal
23          2556   Arachides Decortiquees          végétal
24          2557     Graines de tournesol          végétal
25          2558   Graines Colza/Moutarde          végétal
26          2559         Graines de coton          végétal
27          2560       Coco (Incl Coprah)          végétal
28          2561                   Sésame          végétal
29          2562                Palmistes          végétal
..           ...                      ...              ...
43          2582   Huile de Germe de Maïs          végétal
44          2586  Huil Plantes Oleif Autr          végétal
45          2601                  Tomates          végétal
46          2602                  Oignons          végétal
47          2605          Légumes, Autres          végétal
48          2611      Oranges, Mandarines          végétal
49          2612          Citrons & Limes          végétal
50          2613             Pamplemousse          végétal
51          2614          Agrumes, Autres          végétal
52          2615                  Bananes          végétal
53          2616        Bananes plantains          végétal
54          2617                   Pommes          végétal
55          2618                   Ananas          végétal
56          2619                   Dattes          végétal
57          2620                   Raisin          végétal
58          2625           Fruits, Autres          végétal
59          2630                     Café          végétal
60          2633            Feve de Cacao          végétal
61          2635                      Thé          végétal
62          2640                   Poivre          végétal
63          2641                  Piments          végétal
64          2642                 Girofles          végétal
65          2645           Épices, Autres          végétal
66          2655                      Vin          végétal
67          2656                    Bière          végétal
68          2657       Boissons Fermentés          végétal
69          2658     Boissons Alcooliques          végétal
70          2680    Aliments pour enfants          végétal
71          2745                     Miel          végétal
72          2805         Riz (Eq Blanchi)          végétal

[73 rows x 3 columns]

In [20]:
df = df.merge(df4, how='outer')
df

Code Pays                                    Pays  Année  \
0              1                                 Arménie   2013   
1              2                             Afghanistan   2013   
2              3                                 Albanie   2013   
3              4                                 Algérie   2013   
4              7                                  Angola   2013   
5              8                      Antigua-et-Barbuda   2013   
6              9                               Argentine   2013   
7             10                               Australie   2013   
8             11                                Autriche   2013   
9             12                                 Bahamas   2013   
10            14                                 Barbade   2013   
11            16                              Bangladesh   2013   
12            17                                Bermudes   2013   
13            19         Bolivie (État plurinational de)   2013   
14            20                                Botswana   2013   
15            21                                  Brésil   2013   
16            23                                  Belize   2013   
17            25                            Îles Salomon   2013   
18            26                       Brunéi Darussalam   2013   
19            27                                Bulgarie   2013   
20            28                                 Myanmar   2013   
21            32                                Cameroun   2013   
22            33                                  Canada   2013   
23            35                              Cabo Verde   2013   
24            37               République centrafricaine   2013   
25            38                               Sri Lanka   2013   
26            39                                   Tchad   2013   
27            40                                   Chili   2013   
28            44                                Colombie   2013   
29            46                                   Congo   2013   
...          ...                                     ...    ...   
15295         28                                 Myanmar   2013   
15296         38                               Sri Lanka   2013   
15297         44                                Colombie   2013   
15298         48                              Costa Rica   2013   
15299         50                                  Chypre   2013   
15300         58                                Équateur   2013   
15301         60                             El Salvador   2013   
15302         89                               Guatemala   2013   
15303         93                                   Haïti   2013   
15304         95                                Honduras   2013   
15305        100                                    Inde   2013   
15306        101                               Indonésie   2013   
15307        110                                   Japon   2013   
15308        114                                   Kenya   2013   
15309        120   République démocratique populaire lao   2013   
15310        138                                 Mexique   2013   
15311        149                                   Népal   2013   
15312        157                               Nicaragua   2013   
15313        159                                 Nigéria   2013   
15314        165                                Pakistan   2013   
15315        166                                  Panama   2013   
15316        170                                   Pérou   2013   
15317        171                             Philippines   2013   
15318        215             République-Unie de Tanzanie   2013   
15319        226                                 Ouganda   2013   
15320        236  Venezuela (République bolivarienne du)   2013   
15321        351                                   Chine   2013   
15322         33                                  Cana

In [21]:
a = df[['Pays','Produit','Nature Produit 2','Disponibilité alimentaire en quantité (kg/an)','Disponibilité alimentaire (Kcal/an)']]
a = a[a['Nature Produit 2'] == 'végétal']
a = a.groupby(['Produit']).sum()
a['ratio_energie_poids en Kcal/an'] = a['Disponibilité alimentaire (Kcal/an)'] / a['Disponibilité alimentaire en quantité (kg/an)']
a

Disponibilité alimentaire en quantité (kg/an)  \
Produit                                                                   
Agrumes, Autres                                            1.136492e+07   
Aliments pour enfants                                      7.616444e+05   
Ananas                                                     2.077994e+07   
Arachides Decortiquees                                     1.222103e+07   
Avoine                                                     3.911553e+06   
Bananes                                                    8.594504e+07   
Bananes plantains                                          2.286450e+07   
Bière                                                      1.870574e+08   
Blé                                                        4.578230e+08   
Boissons Alcooliques                                       2.125149e+07   
Boissons Fermentés                                         2.940337e+07   
Café                                                       8.092938e+06   
Citrons & Limes                                            1.418342e+07   
Coco (Incl Coprah)                                         2.223896e+07   
Céréales, Autres                                           5.332189e+06   
Dattes                                                     6.500534e+06   
Edulcorants Autres                                         2.082274e+07   
Feve de Cacao                                              4.764524e+06   
Fruits, Autres                                             1.854805e+08   
Girofles                                                   5.307323e+04   
Graines Colza/Moutarde                                     1.296034e+06   
Graines de coton                                           0.000000e+00   
Graines de tournesol                                       5.574074e+05   
Haricots                                                   1.741721e+07   
Huil Plantes Oleif Autr                                    1.765960e+06   
Huile Graines de Coton                                     3.937959e+06   
Huile d'Arachide                                           3.850252e+06   
Huile d'Olive                                              2.777451e+06   
Huile de Coco                                              2.031975e+06   
Huile de Colza&Moutarde                                    9.807092e+06   
...                                                                 ...   
Millet                                                     2.304001e+07   
Noix                                                       1.640957e+07   
Oignons                                                    7.676720e+07   
Olives                                                     2.756465e+06   
Oranges, Mandarines                                        8.669527e+07   
Orge                                                       6.805964e+06   
Palmistes                                                  4.448000e+03   
Pamplemousse                                               8.291888e+06   
Patates douces                                             5.705984e+07   
Piments                                                    3.138933e+06   
Plantes Oleiferes, Autre                                   9.857597e+05   
Pois                                                       5.841730e+06   
Poivre                                                     4.153251e+05   
Pommes                                                     7.067538e+07   
Pommes de Terre                                            2.390741e+08   
Racines, Nda                                               1.407049e+07   
Raisin                                                     3.211532e+07   
Riz (Eq Blanchi)                                           3.772860e+08   
Seigle                                                     5.510745e+06   
Soja                                                       1.065097e+07   
Sorgho                                             

Renseignez-vous sur la recommandation (de la FAO si possible) concernant le nombre de calories par jour nécessaire à un être humain :
http://www.fao.org/docrep/007/y5686e/y5686e07.htm#bm07.1

    Sedentary or light activity: If this PAL was from a female population, 30 to 50 years old, with mean weight of 55 kg and mean BMR of 5.40 MJ/day (1 290 kcal/day), TEE = 1.53 × 5.40 = 8.26 MJ (1 975 kcal), or 150 kJ (36 kcal)/kg/d.

    Active or moderately active: If this PAL was from a female population, 20 to 25 years old, with mean weight of 57 kg and mean BMR of 5.60 MJ/day (1 338 kcal/day), TEE = 1.76 × 5.60 = 9.86 MJ (2 355 kcal), or 173 kJ (41 kcal)/kg/d.

    Vigorous or vigorously active: If this PAL was from a male population, 20 to 25 years old, with mean weight of 70 kg and mean BMR of 7.30 MJ/day (1 745 kcal/day), TEE = 2.25 × 7.30 = 16.42 MJ (3 925 kcal), or 235 kJ (56 kcal)/kg/d.
    
Nous choisirons 1745 Kcal par jour soit 636 925 Kcal par an

http://www.fao.org/docrep/x8200f/X8200F03.htm
Voir tableau des besoins énergétiques selon sexe, poids et niveau d'acitivité

http://www.fao.org/nouvelle/1998/981204-f.htm
CF Légende carte --> -2000 kcal en rouge

Question : en terme de production de produits végétaux, combien d'humains pourraient être nourris si toute la production végétale était utilisée pour de la nourriture ?

In [22]:
humain_nourris_avec_veg_en_milliers = a['Disponibilité alimentaire (Kcal/an)'].sum() / 636925
humain_nourris_avec_veg_en_milliers

9495789.023495702

Question : en terme de production de produits végétaux, combien d'humains pourraient être nourris si la production végétale, mis à part ses utilisations en Semence, Traitement et Autres utilisation, était utilisée pour de la nourriture ?

In [23]:
a = df[['Pays','Produit','Nature Produit 2','Semences (en milliers de tonnes)','Traitement (en milliers de tonnes)','Autres Utilisations (en milliers de tonnes)','Disponibilité alimentaire en quantité (kg/an)','Disponibilité alimentaire (Kcal/an)']]
a = a[a['Nature Produit 2'] == 'végétal']
a = a.groupby(['Produit']).sum().reset_index()
a['ratio_energie_poids en Kcal/an'] = a['Disponibilité alimentaire (Kcal/an)'] / a['Disponibilité alimentaire en quantité (kg/an)']
a

Produit  Semences (en milliers de tonnes)  \
0            Agrumes, Autres                               0.0   
1      Aliments pour enfants                               0.0   
2                     Ananas                               0.0   
3     Arachides Decortiquees                            1295.0   
4                     Avoine                            2485.0   
5                    Bananes                               0.0   
6          Bananes plantains                               0.0   
7                      Bière                               0.0   
8                        Blé                           34283.0   
9       Boissons Alcooliques                               0.0   
10        Boissons Fermentés                               0.0   
11                      Café                               0.0   
12           Citrons & Limes                               0.0   
13        Coco (Incl Coprah)                             154.0   
14          Céréales, Autres                            1203.0   
15                    Dattes                               0.0   
16        Edulcorants Autres                               0.0   
17             Feve de Cacao                               0.0   
18            Fruits, Autres                               0.0   
19                  Girofles                               0.0   
20    Graines Colza/Moutarde                             695.0   
21          Graines de coton                            1258.0   
22      Graines de tournesol                             666.0   
23                  Haricots                            1227.0   
24   Huil Plantes Oleif Autr                               0.0   
25    Huile Graines de Coton                               0.0   
26          Huile d'Arachide                               0.0   
27             Huile d'Olive                               0.0   
28             Huile de Coco                               0.0   
29   Huile de Colza&Moutarde                               0.0   
..                       ...                               ...   
43                    Millet                             682.0   
44                      Noix                               0.0   
45                   Oignons                               8.0   
46                    Olives                               0.0   
47       Oranges, Mandarines                               0.0   
48                      Orge                            8804.0   
49                 Palmistes                               0.0   
50              Pamplemousse                               0.0   
51            Patates douces                             592.0   
52                   Piments                               0.0   
53  Plantes Oleiferes, Autre                             228.0   
54                      Pois                             838.0   
55                    Poivre                               0.0   
56                    Pommes                               0.0   
57           Pommes de Terre                           32203.0   
58              Racines, Nda                             123.0   
59                    Raisin                               0.0   
60          Riz (Eq Blanchi)                           12194.0   
61                    Seigle                            1262.0   
62                      Soja                            7401.0   
63                    Sorgho                             788.0   
64             Sucre Eq Brut                               0.0   
65      Sucre non centrifugé                               0.0   
66          Sucre, betterave                               0.0   
67              Sucre, canne                           29361.0   
68                    Sésame                              77.0   
69                       Thé                               0.0   
70                   Tomates                               0.0   
71                       Vin                               0.

In [24]:
a['Semences en Kcal/an']= a['Semences (en milliers de tonnes)'] * a['ratio_energie_poids en Kcal/an']
a['Traitement en Kcal/an']= a['Traitement (en milliers de tonnes)'] * a['ratio_energie_poids en Kcal/an']
a['Autres Utilisations en Kcal/an']= a['Autres Utilisations (en milliers de tonnes)'] * a['ratio_energie_poids en Kcal/an']
a['Production Vegetale vers Nourriture'] = a['Semences en Kcal/an'] + a['Traitement en Kcal/an']+a['Autres Utilisations en Kcal/an']+ a['Disponibilité alimentaire (Kcal/an)']
a = a.replace([np.inf, -np.inf], np.nan).dropna(how = 'any') #Suppression valeurs inf et NaN
humain_nourris_prod_en_milliers = a['Production Vegetale vers Nourriture'].sum() /636925
humain_nourris_prod_en_milliers

9502514.659481855

A partir des données téléchargées qui concernent la sous-nutrition, répondez à cette question : Quelle proportion de la population mondiale est considérée comme étant en sous-nutrition ? (Donnez ce résultat pour l'année 2013 ainsi que pour la dernière année disponible).

In [25]:
df2 = pd.read_csv("secu_alim.csv", header = 0) #Lecture du fichier csv avec prise en compte de l'en-tête comme index

In [26]:
df2

Code Domaine                             Domaine  Code zone  \
0             FS  Données de la sécurité alimentaire          2   
1             FS  Données de la sécurité alimentaire          2   
2             FS  Données de la sécurité alimentaire        202   
3             FS  Données de la sécurité alimentaire        202   
4             FS  Données de la sécurité alimentaire          3   
5             FS  Données de la sécurité alimentaire          3   
6             FS  Données de la sécurité alimentaire          4   
7             FS  Données de la sécurité alimentaire          4   
8             FS  Données de la sécurité alimentaire         79   
9             FS  Données de la sécurité alimentaire         79   
10            FS  Données de la sécurité alimentaire          6   
11            FS  Données de la sécurité alimentaire          6   
12            FS  Données de la sécurité alimentaire          7   
13            FS  Données de la sécurité alimentaire          7   
14            FS  Données de la sécurité alimentaire          8   
15            FS  Données de la sécurité alimentaire          8   
16            FS  Données de la sécurité alimentaire        194   
17            FS  Données de la sécurité alimentaire        194   
18            FS  Données de la sécurité alimentaire          9   
19            FS  Données de la sécurité alimentaire          9   
20            FS  Données de la sécurité alimentaire          1   
21            FS  Données de la sécurité alimentaire          1   
22            FS  Données de la sécurité alimentaire         10   
23            FS  Données de la sécurité alimentaire         10   
24            FS  Données de la sécurité alimentaire         11   
25            FS  Données de la sécurité alimentaire         11   
26            FS  Données de la sécurité alimentaire         52   
27            FS  Données de la sécurité alimentaire         52   
28            FS  Données de la sécurité alimentaire         12   
29            FS  Données de la sécurité alimentaire         12   
..           ...                                 ...        ...   
370           FS  Données de la sécurité alimentaire        218   
371           FS  Données de la sécurité alimentaire        218   
372           FS  Données de la sécurité alimentaire        219   
373           FS  Données de la sécurité alimentaire        219   
374           FS  Données de la sécurité alimentaire        220   
375           FS  Données de la sécurité alimentaire        220   
376           FS  Données de la sécurité alimentaire        222   
377           FS  Données de la sécurité alimentaire        222   
378           FS  Données de la sécurité alimentaire        213   
379           FS  Données de la sécurité alimentaire        213   
380           FS  Données de la sécurité alimentaire        223   
381           FS  Données de la sécurité alimentaire        223   
382           FS  Données de la sécurité alimentaire        227   
383           FS  Données de la sécurité alimentaire        227   
384           FS  Données de la sécurité alimentaire        230   
385           FS  Données de la sécurité alimentaire        230   
386           FS  Données de la sécurité alimentaire        234   
387           FS  Données de la sécurité alimentaire        234   
388           FS  Données de la sécurité alimentaire        155   
389           FS  Données de la sécurité alimentaire        155   
390           FS  Données de la sécurité alimentaire        236   
391           FS  Données de la sécurité alimentaire        236   
392           FS  Données de la sécurité alimentaire        237   
393           FS  Données de la sécurité alimentaire        237   
394           FS  Données de la sécurité alimentaire        249   
395           FS  Données de la sécurité alimentaire        249   
396           FS  Données de la sécurité alimentaire        251   
397           FS  Données de la sécurité alimentair

In [27]:
a = df2[['Code zone','Zone','Valeur','Code année']]
a = a.groupby(['Code zone', 'Zone','Code année']).sum().reset_index()
a

Code zone                                    Zone  Code année  Valeur
0            1                                 Arménie    20122014     0.1
1            1                                 Arménie    20142016     0.1
2            2                             Afghanistan    20122014     7.2
3            2                             Afghanistan    20142016     7.5
4            3                                 Albanie    20122014     0.2
5            3                                 Albanie    20142016     0.1
6            4                                 Algérie    20122014     1.8
7            4                                 Algérie    20142016     1.8
8            5                       Samoa américaines    20122014     0.0
9            5                       Samoa américaines    20142016     0.0
10           6                                 Andorre    20122014     0.0
11           6                                 Andorre    20142016     0.0
12           7                                  Angola    20122014     3.6
13           7                                  Angola    20142016     3.5
14           8                      Antigua-et-Barbuda    20122014     0.0
15           8                      Antigua-et-Barbuda    20142016     0.0
16           9                               Argentine    20122014     1.5
17           9                               Argentine    20142016     1.6
18          10                               Australie    20122014     0.0
19          10                               Australie    20142016     0.0
20          11                                Autriche    20122014     0.0
21          11                                Autriche    20142016     0.0
22          12                                 Bahamas    20122014     0.0
23          12                                 Bahamas    20142016     0.0
24          13                                 Bahreïn    20122014     0.0
25          13                                 Bahreïn    20142016     0.0
26          14                                 Barbade    20122014     0.0
27          14                                 Barbade    20142016     0.0
28          16                              Bangladesh    20122014    25.8
29          16                              Bangladesh    20142016    24.4
..         ...                                     ...         ...     ...
370        236  Venezuela (République bolivarienne du)    20122014     2.0
371        236  Venezuela (République bolivarienne du)    20142016     4.1
372        237                                Viet Nam    20122014    10.8
373        237                                Viet Nam    20142016    10.0
374        238                                Éthiopie    20122014    27.8
375        238                                Éthiopie    20142016    28.6
376        244                                   Samoa    20122014     0.0
377        244                                   Samoa    20142016     0.0
378        249                                   Yémen    20122014     6.3
379        249                                   Yémen    20142016     7.7
380        250        République démocratique du Congo    20122014     0.0
381        250        République démocratique du Congo    20142016     0.0
382        251                                  Zambie    20122014     7.2
383        251                                  Zambie    20142016     7.4
384        255                                Belgique    20122014     0.0
385        255                                Belgique    20142016     0.0
386        256                              Luxembourg    20122014     0.0
387        256                              Luxembourg    20142016     0.0
388        272                                  Serbie    20122014     0.5
389        272                                  Serbie    20142016     0.5
390        273                              Monténégro    20122014     0.0
391        273                

In [28]:
#a = df_population.merge(a, left_on = ['Code Pays'], right_on = ['Code zone'])
b = a[a['Code année'] == 20122014]
population_sous_nourri_2013 = b['Valeur'].sum()*1000000
population_sous_nourri_2013

722900000.0000001

In [29]:
proportion_population_sous_alim_dans_population_2013 = (population_sous_nourri_2013 / (population_mondiale_2013_en_milliers * 1000)) *100
proportion_population_sous_alim_dans_population_2013

10.331089333268167

In [30]:
a = df2[['Code zone','Zone','Valeur','Code année']]
a = a.groupby(['Code zone', 'Zone','Code année']).sum().reset_index()

c = a[a['Code année'] == 20142016]
population_sous_nourri_2015 = c['Valeur'].sum()*1000000
population_sous_nourri_2015

728600000.0

http://economiedurable.over-blog.com/article-la-population-mondiale-au-premier-janvier-2015-125282595.html
population mondiale en 2015 = 7 260 milliards (moyenne)

In [31]:
population_mondiale_2015 = 7260000000
proportion_population_sous_alim_dans_population_2015 = (population_sous_nourri_2015 / population_mondiale_2015) *100
proportion_population_sous_alim_dans_population_2015

10.03581267217631

Établissez la liste des produits (ainsi que leur code) considéré comme des céréales selon la FAO.

In [32]:
df3 = pd.read_csv("cereales.csv", header = 0) #Lecture du fichier csv avec prise en compte de l'en-tête comme index
df3 = df3.groupby(['Code Produit', 'Produit']).sum().reset_index()
df3 = df3[['Code Produit','Produit']]
df3

Code Produit           Produit
0          2511               Blé
1          2513              Orge
2          2514              Maïs
3          2515            Seigle
4          2516            Avoine
5          2517            Millet
6          2518            Sorgho
7          2520  Céréales, Autres
8          2805  Riz (Eq Blanchi)

Sélectionnez (filtrez) dans vos données les informations concernant les céréales qui partent pour l'alimentation animale (en anglais : Feed).

In [33]:
a = df[['Pays','Produit','Nature Produit','Aliments pour animaux (en milliers de tonnes)','Disponibilité alimentaire en quantité (kg/an)','Disponibilité alimentaire (Kcal/an)']]
a = a[a['Nature Produit'] == 'céréales']
a = a.groupby(['Produit']).sum().reset_index()
a['ratio_energie_poids en Kcal/an'] = a['Disponibilité alimentaire (Kcal/an)'] / a['Disponibilité alimentaire en quantité (kg/an)']
a

Produit  Aliments pour animaux (en milliers de tonnes)  \
0            Avoine                                        16251.0   
1               Blé                                       129668.0   
2  Céréales, Autres                                        19036.0   
3              Maïs                                       546116.0   
4            Millet                                         3306.0   
5              Orge                                        92658.0   
6  Riz (Eq Blanchi)                                        33594.0   
7            Seigle                                         8099.0   
8            Sorgho                                        24808.0   

   Disponibilité alimentaire en quantité (kg/an)  \
0                                   3.911553e+06   
1                                   4.578230e+08   
2                                   5.332189e+06   
3                                   1.251851e+08   
4                                   2.304001e+07   
5                                   6.805964e+06   
6                                   3.772860e+08   
7                                   5.510745e+06   
8                                   2.415447e+07   

   Disponibilité alimentaire (Kcal/an)  ratio_energie_poids en Kcal/an  
0                         7.337340e+09                     1875.812206  
1                         1.345330e+12                     2938.537731  
2                         1.648755e+10                     3092.079337  
3                         3.766174e+11                     3008.484894  
4                         6.874463e+10                     2983.706786  
5                         1.762427e+10                     2589.533208  
6                         1.381105e+12                     3660.632099  
7                         1.544829e+10                     2803.303398  
8                         7.251544e+10                     3002.153376

A l'aide de la table donnant les ratios énergie/poids de chaque aliment (que vous avez établie précédemment), convertir ces données (filtrées) en kcal.

In [34]:
a['Aliments pour animaux en Kcal/an']= a['Aliments pour animaux (en milliers de tonnes)'] * a['ratio_energie_poids en Kcal/an']
a

Produit  Aliments pour animaux (en milliers de tonnes)  \
0            Avoine                                        16251.0   
1               Blé                                       129668.0   
2  Céréales, Autres                                        19036.0   
3              Maïs                                       546116.0   
4            Millet                                         3306.0   
5              Orge                                        92658.0   
6  Riz (Eq Blanchi)                                        33594.0   
7            Seigle                                         8099.0   
8            Sorgho                                        24808.0   

   Disponibilité alimentaire en quantité (kg/an)  \
0                                   3.911553e+06   
1                                   4.578230e+08   
2                                   5.332189e+06   
3                                   1.251851e+08   
4                                   2.304001e+07   
5                                   6.805964e+06   
6                                   3.772860e+08   
7                                   5.510745e+06   
8                                   2.415447e+07   

   Disponibilité alimentaire (Kcal/an)  ratio_energie_poids en Kcal/an  \
0                         7.337340e+09                     1875.812206   
1                         1.345330e+12                     2938.537731   
2                         1.648755e+10                     3092.079337   
3                         3.766174e+11                     3008.484894   
4                         6.874463e+10                     2983.706786   
5                         1.762427e+10                     2589.533208   
6                         1.381105e+12                     3660.632099   
7                         1.544829e+10                     2803.303398   
8                         7.251544e+10                     3002.153376   

   Aliments pour animaux en Kcal/an  
0                      3.048382e+07  
1                      3.810343e+08  
2                      5.886082e+07  
3                      1.642982e+09  
4                      9.864135e+06  
5                      2.399410e+08  
6                      1.229753e+08  
7                      2.270395e+07  
8                      7.447742e+07

De la même manière que vous avez calculé la table des ratios énergie/poids , calculez les ratios poids de protéines/poids total, puis convertissez ces données (filtrées) en terme de protéines (unité : gramme).

In [35]:
df


Code Pays                                    Pays  Année  \
0              1                                 Arménie   2013   
1              2                             Afghanistan   2013   
2              3                                 Albanie   2013   
3              4                                 Algérie   2013   
4              7                                  Angola   2013   
5              8                      Antigua-et-Barbuda   2013   
6              9                               Argentine   2013   
7             10                               Australie   2013   
8             11                                Autriche   2013   
9             12                                 Bahamas   2013   
10            14                                 Barbade   2013   
11            16                              Bangladesh   2013   
12            17                                Bermudes   2013   
13            19         Bolivie (État plurinational de)   2013   
14            20                                Botswana   2013   
15            21                                  Brésil   2013   
16            23                                  Belize   2013   
17            25                            Îles Salomon   2013   
18            26                       Brunéi Darussalam   2013   
19            27                                Bulgarie   2013   
20            28                                 Myanmar   2013   
21            32                                Cameroun   2013   
22            33                                  Canada   2013   
23            35                              Cabo Verde   2013   
24            37               République centrafricaine   2013   
25            38                               Sri Lanka   2013   
26            39                                   Tchad   2013   
27            40                                   Chili   2013   
28            44                                Colombie   2013   
29            46                                   Congo   2013   
...          ...                                     ...    ...   
15295         28                                 Myanmar   2013   
15296         38                               Sri Lanka   2013   
15297         44                                Colombie   2013   
15298         48                              Costa Rica   2013   
15299         50                                  Chypre   2013   
15300         58                                Équateur   2013   
15301         60                             El Salvador   2013   
15302         89                               Guatemala   2013   
15303         93                                   Haïti   2013   
15304         95                                Honduras   2013   
15305        100                                    Inde   2013   
15306        101                               Indonésie   2013   
15307        110                                   Japon   2013   
15308        114                                   Kenya   2013   
15309        120   République démocratique populaire lao   2013   
15310        138                                 Mexique   2013   
15311        149                                   Népal   2013   
15312        157                               Nicaragua   2013   
15313        159                                 Nigéria   2013   
15314        165                                Pakistan   2013   
15315        166                                  Panama   2013   
15316        170                                   Pérou   2013   
15317        171                             Philippines   2013   
15318        215             République-Unie de Tanzanie   2013   
15319        226                                 Ouganda   2013   
15320        236  Venezuela (République bolivarienne du)   2013   
15321        351                                   Chine   2013   
15322         33                                  Cana

In [36]:
df['Disponibilité de protéines en quantité (Kg/an)'] = df['Disponibilité de protéines en quantité (g/personne/jour)'] *365 * df['Population totale (en milliers)'] / 1000
df['ratio_poids_proteines_poids_total_en_kg_an'] = df['Disponibilité de protéines en quantité (Kg/an)'] / df['Disponibilité alimentaire en quantité (kg/an)'] 

In [37]:
df

Code Pays                                    Pays  Année  \
0              1                                 Arménie   2013   
1              2                             Afghanistan   2013   
2              3                                 Albanie   2013   
3              4                                 Algérie   2013   
4              7                                  Angola   2013   
5              8                      Antigua-et-Barbuda   2013   
6              9                               Argentine   2013   
7             10                               Australie   2013   
8             11                                Autriche   2013   
9             12                                 Bahamas   2013   
10            14                                 Barbade   2013   
11            16                              Bangladesh   2013   
12            17                                Bermudes   2013   
13            19         Bolivie (État plurinational de)   2013   
14            20                                Botswana   2013   
15            21                                  Brésil   2013   
16            23                                  Belize   2013   
17            25                            Îles Salomon   2013   
18            26                       Brunéi Darussalam   2013   
19            27                                Bulgarie   2013   
20            28                                 Myanmar   2013   
21            32                                Cameroun   2013   
22            33                                  Canada   2013   
23            35                              Cabo Verde   2013   
24            37               République centrafricaine   2013   
25            38                               Sri Lanka   2013   
26            39                                   Tchad   2013   
27            40                                   Chili   2013   
28            44                                Colombie   2013   
29            46                                   Congo   2013   
...          ...                                     ...    ...   
15295         28                                 Myanmar   2013   
15296         38                               Sri Lanka   2013   
15297         44                                Colombie   2013   
15298         48                              Costa Rica   2013   
15299         50                                  Chypre   2013   
15300         58                                Équateur   2013   
15301         60                             El Salvador   2013   
15302         89                               Guatemala   2013   
15303         93                                   Haïti   2013   
15304         95                                Honduras   2013   
15305        100                                    Inde   2013   
15306        101                               Indonésie   2013   
15307        110                                   Japon   2013   
15308        114                                   Kenya   2013   
15309        120   République démocratique populaire lao   2013   
15310        138                                 Mexique   2013   
15311        149                                   Népal   2013   
15312        157                               Nicaragua   2013   
15313        159                                 Nigéria   2013   
15314        165                                Pakistan   2013   
15315        166                                  Panama   2013   
15316        170                                   Pérou   2013   
15317        171                             Philippines   2013   
15318        215             République-Unie de Tanzanie   2013   
15319        226                                 Ouganda   2013   
15320        236  Venezuela (République bolivarienne du)   2013   
15321        351                                   Chine   2013   
15322         33                                  Cana

In [38]:
b = df[['Pays','Produit','Nature Produit','Aliments pour animaux (en milliers de tonnes)','Disponibilité alimentaire en quantité (kg/an)','Disponibilité de protéines en quantité (Kg/an)']]
b = b[b['Nature Produit'] == 'céréales']
b = b.groupby(['Produit']).sum().reset_index()
b['ratio_poids_proteines_poids_total_en_kg_an'] = b['Disponibilité de protéines en quantité (Kg/an)'] / b['Disponibilité alimentaire en quantité (kg/an)']
b

Produit  Aliments pour animaux (en milliers de tonnes)  \
0            Avoine                                        16251.0   
1               Blé                                       129668.0   
2  Céréales, Autres                                        19036.0   
3              Maïs                                       546116.0   
4            Millet                                         3306.0   
5              Orge                                        92658.0   
6  Riz (Eq Blanchi)                                        33594.0   
7            Seigle                                         8099.0   
8            Sorgho                                        24808.0   

   Disponibilité alimentaire en quantité (kg/an)  \
0                                   3.911553e+06   
1                                   4.578230e+08   
2                                   5.332189e+06   
3                                   1.251851e+08   
4                                   2.304001e+07   
5                                   6.805964e+06   
6                                   3.772860e+08   
7                                   5.510745e+06   
8                                   2.415447e+07   

   Disponibilité de protéines en quantité (Kg/an)  \
0                                    3.212602e+05   
1                                    4.053949e+07   
2                                    4.552884e+05   
3                                    9.151895e+06   
4                                    1.820736e+06   
5                                    4.942710e+05   
6                                    2.588200e+07   
7                                    4.047473e+05   
8                                    2.149033e+06   

   ratio_poids_proteines_poids_total_en_kg_an  
0                                    0.082131  
1                                    0.088548  
2                                    0.085385  
3                                    0.073107  
4                                    0.079025  
5                                    0.072623  
6                                    0.068600  
7                                    0.073447  
8                                    0.088970

In [39]:
b['Aliments pour animaux en poids protéines/an']= b['Aliments pour animaux (en milliers de tonnes)'] * b['ratio_poids_proteines_poids_total_en_kg_an']
b

Produit  Aliments pour animaux (en milliers de tonnes)  \
0            Avoine                                        16251.0   
1               Blé                                       129668.0   
2  Céréales, Autres                                        19036.0   
3              Maïs                                       546116.0   
4            Millet                                         3306.0   
5              Orge                                        92658.0   
6  Riz (Eq Blanchi)                                        33594.0   
7            Seigle                                         8099.0   
8            Sorgho                                        24808.0   

   Disponibilité alimentaire en quantité (kg/an)  \
0                                   3.911553e+06   
1                                   4.578230e+08   
2                                   5.332189e+06   
3                                   1.251851e+08   
4                                   2.304001e+07   
5                                   6.805964e+06   
6                                   3.772860e+08   
7                                   5.510745e+06   
8                                   2.415447e+07   

   Disponibilité de protéines en quantité (Kg/an)  \
0                                    3.212602e+05   
1                                    4.053949e+07   
2                                    4.552884e+05   
3                                    9.151895e+06   
4                                    1.820736e+06   
5                                    4.942710e+05   
6                                    2.588200e+07   
7                                    4.047473e+05   
8                                    2.149033e+06   

   ratio_poids_proteines_poids_total_en_kg_an  \
0                                    0.082131   
1                                    0.088548   
2                                    0.085385   
3                                    0.073107   
4                                    0.079025   
5                                    0.072623   
6                                    0.068600   
7                                    0.073447   
8                                    0.088970   

   Aliments pour animaux en poids protéines/an  
0                                  1334.712639  
1                                 11481.894463  
2                                  1625.386910  
3                                 39924.859644  
4                                   261.256554  
5                                  6729.121404  
6                                  2304.564311  
7                                   594.846679  
8                                  2207.177805

Question : quelle proportion (en terme de poids) de la disponibilité mondiale de céréales est destinée à l'alimentation animale ?

In [40]:
c = df[['Pays','Produit','Nature Produit','Aliments pour animaux (en milliers de tonnes)','Disponibilité intérieure (en milliers de tonnes)']]
c = c[c['Nature Produit'] == 'céréales']
c = c.groupby(['Produit']).sum().reset_index()
c

Produit  Aliments pour animaux (en milliers de tonnes)  \
0            Avoine                                        16251.0   
1               Blé                                       129668.0   
2  Céréales, Autres                                        19036.0   
3              Maïs                                       546116.0   
4            Millet                                         3306.0   
5              Orge                                        92658.0   
6  Riz (Eq Blanchi)                                        33594.0   
7            Seigle                                         8099.0   
8            Sorgho                                        24808.0   

   Disponibilité intérieure (en milliers de tonnes)  
0                                           23407.0  
1                                          679497.0  
2                                           27485.0  
3                                          955798.0  
4                                           29911.0  
5                                          140439.0  
6                                          475656.0  
7                                           16567.0  
8                                           58237.0

In [41]:
proportion_alim_ani_dans_dispo_int = c['Aliments pour animaux (en milliers de tonnes)'].sum() / c['Disponibilité intérieure (en milliers de tonnes)'].sum()
proportion_alim_ani_dans_dispo_int

0.3629152840655805

Avec l'aide des tables énergie/poids et poids de protéines/poids total, calculez la disponibilité mondiale totale de produit animaux, que vous exprimerez en kcal et en grammes de protéines.

In [42]:
df['Nature Produit 2'].replace([np.nan], 'animal', inplace=True)

In [43]:
df

Code Pays                                    Pays  Année  \
0              1                                 Arménie   2013   
1              2                             Afghanistan   2013   
2              3                                 Albanie   2013   
3              4                                 Algérie   2013   
4              7                                  Angola   2013   
5              8                      Antigua-et-Barbuda   2013   
6              9                               Argentine   2013   
7             10                               Australie   2013   
8             11                                Autriche   2013   
9             12                                 Bahamas   2013   
10            14                                 Barbade   2013   
11            16                              Bangladesh   2013   
12            17                                Bermudes   2013   
13            19         Bolivie (État plurinational de)   2013   
14            20                                Botswana   2013   
15            21                                  Brésil   2013   
16            23                                  Belize   2013   
17            25                            Îles Salomon   2013   
18            26                       Brunéi Darussalam   2013   
19            27                                Bulgarie   2013   
20            28                                 Myanmar   2013   
21            32                                Cameroun   2013   
22            33                                  Canada   2013   
23            35                              Cabo Verde   2013   
24            37               République centrafricaine   2013   
25            38                               Sri Lanka   2013   
26            39                                   Tchad   2013   
27            40                                   Chili   2013   
28            44                                Colombie   2013   
29            46                                   Congo   2013   
...          ...                                     ...    ...   
15295         28                                 Myanmar   2013   
15296         38                               Sri Lanka   2013   
15297         44                                Colombie   2013   
15298         48                              Costa Rica   2013   
15299         50                                  Chypre   2013   
15300         58                                Équateur   2013   
15301         60                             El Salvador   2013   
15302         89                               Guatemala   2013   
15303         93                                   Haïti   2013   
15304         95                                Honduras   2013   
15305        100                                    Inde   2013   
15306        101                               Indonésie   2013   
15307        110                                   Japon   2013   
15308        114                                   Kenya   2013   
15309        120   République démocratique populaire lao   2013   
15310        138                                 Mexique   2013   
15311        149                                   Népal   2013   
15312        157                               Nicaragua   2013   
15313        159                                 Nigéria   2013   
15314        165                                Pakistan   2013   
15315        166                                  Panama   2013   
15316        170                                   Pérou   2013   
15317        171                             Philippines   2013   
15318        215             République-Unie de Tanzanie   2013   
15319        226                                 Ouganda   2013   
15320        236  Venezuela (République bolivarienne du)   2013   
15321        351                                   Chine   2013   
15322         33                                  Cana

In [44]:
df5 = pd.read_csv("viandes.csv", header = 0) #Lecture du fichier csv avec prise en compte de l'en-tête comme index
df5 = df5.groupby(['Code Produit', 'Produit']).sum().reset_index()
df5 = df5[['Code Produit','Produit']]
df5['Nature Produit 3'] = 'viandes'
df5

Code Produit                 Produit Nature Produit 3
0          2731        Viande de Bovins          viandes
1          2732  Viande d'Ovins/Caprins          viandes
2          2733        Viande de Suides          viandes
3          2734     Viande de Volailles          viandes
4          2735           Viande, Autre          viandes

In [64]:
df = df.merge(df5, how='outer')
df['Nature Produit 3'].replace([np.nan], 'non viande', inplace=True)
df
#df.to_csv('df_base_projet3.csv', index_label=False)

In [46]:
c = df[['Pays','Produit','Nature Produit 2','Disponibilité alimentaire (Kcal/an)','Disponibilité de protéines en quantité (Kg/an)']]
c = c[c['Nature Produit 2'] == 'animal']
c = c.groupby(['Produit']).sum().reset_index()
c

Produit  Disponibilité alimentaire (Kcal/an)  \
0           Abats Comestible                         1.756899e+10   
1     Alcool, non Comestible                         0.000000e+00   
2   Animaux Aquatiques Autre                         5.599158e+08   
3               Beurre, Ghee                         7.629615e+10   
4               Cephalopodes                         2.781070e+09   
5                  Crustacés                         5.490460e+09   
6                      Crème                         5.692490e+09   
7     Graisses Animales Crue                         7.391450e+10   
8   Huiles de Foie de Poisso                         1.000845e+08   
9         Huiles de Poissons                         6.076987e+08   
10        Lait - Excl Beurre                         3.517248e+11   
11              Miscellanees                         2.501126e+09   
12        Mollusques, Autres                         3.049108e+09   
13                     Oeufs                         9.169544e+10   
14                 Perciform                         1.419630e+10   
15        Plantes Aquatiques                         5.095741e+09   
16        Poissons Eau Douce                         3.318129e+10   
17   Poissons Marins, Autres                         6.243194e+09   
18       Poissons Pelagiques                         2.093399e+10   
19    Viande d'Ovins/Caprins                         2.756593e+10   
20    Viande de Anim Aquatiq                         0.000000e+00   
21          Viande de Bovins                         1.011677e+11   
22          Viande de Suides                         3.178415e+11   
23       Viande de Volailles                         1.500870e+11   
24             Viande, Autre                         8.037495e+09   

    Disponibilité de protéines en quantité (Kg/an)  
0                                     2.795892e+06  
1                                     0.000000e+00  
2                                     4.637247e+04  
3                                     6.315940e+04  
4                                     5.036496e+05  
5                                     1.115226e+06  
6                                     8.397421e+04  
7                                     6.444460e+04  
8                                     0.000000e+00  
9                                     0.000000e+00  
10                                    2.100778e+07  
11                                    9.503382e+04  
12                                    4.401758e+05  
13                                    7.136641e+06  
14                                    2.178925e+06  
15                                    3.921758e+05  
16                                    5.221458e+06  
17                                    9.775220e+05  
18                                    2.902363e+06  
19                                    1.847171e+06  
20                                    0.000000e+00  
21                                    9.031706e+06  
22                                    1.186332e+07  
23                                    1.317468e+07  
24                                    1.211730e+06

In [47]:
disponibilite_mondiale_totale_produit_animaux_kcal = c['Disponibilité alimentaire (Kcal/an)'].sum()
disponibilite_mondiale_totale_produit_animaux_kcal

1316331929885.0

In [48]:
disponibilite_mondiale_totale_produit_animaux_g_de_prot = c['Disponibilité de protéines en quantité (Kg/an)'].sum()*1000
disponibilite_mondiale_totale_produit_animaux_g_de_prot

82153399394.4

Calculez les 3 valeurs du ratio suivant, lorsque les quantités sont exprimées en kg, puis en kcal, puis en grammes de protéines :

Disponibilité de céréales destinée à l'alimentation animale / Disponibilité animale destinée à l'alimentation humaine

In [49]:
d = df[['Pays','Produit','Nature Produit','Nature Produit 2','Nature Produit 3','Aliments pour animaux (en milliers de tonnes)','Nourriture (en milliers de tonnes)','Disponibilité alimentaire (Kcal/an)','Disponibilité de protéines en quantité (Kg/an)','Disponibilité alimentaire en quantité (kg/an)']]
d = d.groupby(['Produit','Nature Produit', 'Nature Produit 2', 'Nature Produit 3']).sum().reset_index()
d['ratio_poids_proteines_poids_total_en_kg_an'] = d['Disponibilité de protéines en quantité (Kg/an)'] / d['Disponibilité alimentaire en quantité (kg/an)']
d['ratio_energie_poids en Kcal/an'] = d['Disponibilité alimentaire (Kcal/an)'] / d['Disponibilité alimentaire en quantité (kg/an)']
d

Produit Nature Produit Nature Produit 2 Nature Produit 3  \
0           Abats Comestible          autre           animal       non viande   
1            Agrumes, Autres          autre          végétal       non viande   
2     Alcool, non Comestible          autre           animal       non viande   
3      Aliments pour enfants          autre          végétal       non viande   
4                     Ananas          autre          végétal       non viande   
5   Animaux Aquatiques Autre          autre           animal       non viande   
6     Arachides Decortiquees          autre          végétal       non viande   
7                     Avoine       céréales          végétal       non viande   
8                    Bananes          autre          végétal       non viande   
9          Bananes plantains          autre          végétal       non viande   
10              Beurre, Ghee          autre           animal       non viande   
11                     Bière          autre          végétal       non viande   
12                       Blé       céréales          végétal       non viande   
13      Boissons Alcooliques          autre          végétal       non viande   
14        Boissons Fermentés          autre          végétal       non viande   
15                      Café          autre          végétal       non viande   
16              Cephalopodes          autre           animal       non viande   
17           Citrons & Limes          autre          végétal       non viande   
18        Coco (Incl Coprah)          autre          végétal       non viande   
19                 Crustacés          autre           animal       non viande   
20                     Crème          autre           animal       non viande   
21          Céréales, Autres       céréales          végétal       non viande   
22                    Dattes          autre          végétal       non viande   
23        Edulcorants Autres          autre          végétal       non viande   
24             Feve de Cacao          autre          végétal       non viande   
25            Fruits, Autres          autre          végétal       non viande   
26                  Girofles          autre          végétal       non viande   
27    Graines Colza/Moutarde          autre          végétal       non viande   
28          Graines de coton          autre          végétal       non viande   
29      Graines de tournesol          autre          végétal       non viande   
..                       ...            ...              ...              ...   
68        Plantes Aquatiques          autre           animal       non viande   
69  Plantes Oleiferes, Autre          autre          végétal       non viande   
70                      Pois          autre          végétal       non viande   
71        Poissons Eau Douce          autre           animal       non viande   
72   Poissons Marins, Autres          autre           animal       non viande   
73       Poissons Pelagiques          autre           animal       non viande   
74                    Poivre          autre          végétal       non viande   
75                    Pommes          autre          végétal       non viande   
76           Pommes de Terre          autre          végétal       non viande   
77              Racines, Nda          autre          végétal       non viande   
78                    Raisin          autre          végétal       non viande   
79          Riz (Eq Blanchi)       céréales          végétal       non viande   
80                    Seigle       céréales          végétal       non viande   
81                      Soja          autre          végétal       non viande   
82                    Sorgho       céréales          végétal       non viande   
83             Sucre Eq Brut          autre          végétal       non viande   
84      Sucre non centrifugé          autre          végétal       non viande   
85          Sucre, betterave          autre         

In [50]:
d['Aliments pour animaux en poids protéines/an']= d['Aliments pour animaux (en milliers de tonnes)'] * d['ratio_poids_proteines_poids_total_en_kg_an']
d['Aliments pour animaux en Kcal/an']= d['Aliments pour animaux (en milliers de tonnes)'] * d['ratio_energie_poids en Kcal/an']
d

Produit Nature Produit Nature Produit 2 Nature Produit 3  \
0           Abats Comestible          autre           animal       non viande   
1            Agrumes, Autres          autre          végétal       non viande   
2     Alcool, non Comestible          autre           animal       non viande   
3      Aliments pour enfants          autre          végétal       non viande   
4                     Ananas          autre          végétal       non viande   
5   Animaux Aquatiques Autre          autre           animal       non viande   
6     Arachides Decortiquees          autre          végétal       non viande   
7                     Avoine       céréales          végétal       non viande   
8                    Bananes          autre          végétal       non viande   
9          Bananes plantains          autre          végétal       non viande   
10              Beurre, Ghee          autre           animal       non viande   
11                     Bière          autre          végétal       non viande   
12                       Blé       céréales          végétal       non viande   
13      Boissons Alcooliques          autre          végétal       non viande   
14        Boissons Fermentés          autre          végétal       non viande   
15                      Café          autre          végétal       non viande   
16              Cephalopodes          autre           animal       non viande   
17           Citrons & Limes          autre          végétal       non viande   
18        Coco (Incl Coprah)          autre          végétal       non viande   
19                 Crustacés          autre           animal       non viande   
20                     Crème          autre           animal       non viande   
21          Céréales, Autres       céréales          végétal       non viande   
22                    Dattes          autre          végétal       non viande   
23        Edulcorants Autres          autre          végétal       non viande   
24             Feve de Cacao          autre          végétal       non viande   
25            Fruits, Autres          autre          végétal       non viande   
26                  Girofles          autre          végétal       non viande   
27    Graines Colza/Moutarde          autre          végétal       non viande   
28          Graines de coton          autre          végétal       non viande   
29      Graines de tournesol          autre          végétal       non viande   
..                       ...            ...              ...              ...   
68        Plantes Aquatiques          autre           animal       non viande   
69  Plantes Oleiferes, Autre          autre          végétal       non viande   
70                      Pois          autre          végétal       non viande   
71        Poissons Eau Douce          autre           animal       non viande   
72   Poissons Marins, Autres          autre           animal       non viande   
73       Poissons Pelagiques          autre           animal       non viande   
74                    Poivre          autre          végétal       non viande   
75                    Pommes          autre          végétal       non viande   
76           Pommes de Terre          autre          végétal       non viande   
77              Racines, Nda          autre          végétal       non viande   
78                    Raisin          autre          végétal       non viande   
79          Riz (Eq Blanchi)       céréales          végétal       non viande   
80                    Seigle       céréales          végétal       non viande   
81                      Soja          autre          végétal       non viande   
82                    Sorgho       céréales          végétal       non viande   
83             Sucre Eq Brut          autre          végétal       non viande   
84      Sucre non centrifugé          autre          végétal       non viande   
85          Sucre, betterave          autre         

In [51]:
e = (d.loc[(d['Nature Produit']=="céréales") & (d['Nature Produit 2'] =="végétal"),:])
#e = e.replace([np.inf, -np.inf], 0)
e

Produit Nature Produit Nature Produit 2 Nature Produit 3  \
7             Avoine       céréales          végétal       non viande   
12               Blé       céréales          végétal       non viande   
21  Céréales, Autres       céréales          végétal       non viande   
52              Maïs       céréales          végétal       non viande   
54            Millet       céréales          végétal       non viande   
62              Orge       céréales          végétal       non viande   
79  Riz (Eq Blanchi)       céréales          végétal       non viande   
80            Seigle       céréales          végétal       non viande   
82            Sorgho       céréales          végétal       non viande   

    Aliments pour animaux (en milliers de tonnes)  \
7                                         16251.0   
12                                       129668.0   
21                                        19036.0   
52                                       546116.0   
54                                         3306.0   
62                                        92658.0   
79                                        33594.0   
80                                         8099.0   
82                                        24808.0   

    Nourriture (en milliers de tonnes)  Disponibilité alimentaire (Kcal/an)  \
7                               3904.0                         7.337340e+09   
12                            457824.0                         1.345330e+12   
21                              5324.0                         1.648755e+10   
52                            125184.0                         3.766174e+11   
54                             23040.0                         6.874463e+10   
62                              6795.0                         1.762427e+10   
79                            377286.0                         1.381105e+12   
80                              5502.0                         1.544829e+10   
82                             24153.0                         7.251544e+10   

    Disponibilité de protéines en quantité (Kg/an)  \
7                                     3.212602e+05   
12                                    4.053949e+07   
21                                    4.552884e+05   
52                                    9.151895e+06   
54                                    1.820736e+06   
62                                    4.942710e+05   
79                                    2.588200e+07   
80                                    4.047473e+05   
82                                    2.149033e+06   

    Disponibilité alimentaire en quantité (kg/an)  \
7                                    3.911553e+06   
12                                   4.578230e+08   
21                                   5.332189e+06   
52                                   1.251851e+08   
54                                   2.304001e+07   
62                                   6.805964e+06   
79                                   3.772860e+08   
80                                   5.510745e+06   
82                                   2.415447e+07   

    ratio_poids_proteines_poids_total_en_kg_an  \
7                                     0.082131   
12                                    0.088548   
21                                    0.085385   
52                                    0.073107   
54                                    0.079025   
62                                    0.072623   
79                                    0.068600   
80                                    0.073447   
82                                    0.088970   

    ratio_energie_poids en Kcal/an  \
7                      1875.812206   
12                     2938.537731   
21                     3092.079337   
52                     3008.484894   
54                     2983.706786   
62                     2589.533208   
79                     3660.632099   
80                     2803.303398   
82                     3002.153376   

    Al

In [52]:
dispo_cereales_destinee_alim_ani_kg_an = e['Aliments pour animaux (en milliers de tonnes)'].sum()
dispo_cereales_destinee_alim_ani_kg_an

873536.0

In [53]:
dispo_cereales_destinee_alim_ani_kcal_an = e['Aliments pour animaux en Kcal/an'].sum()
dispo_cereales_destinee_alim_ani_kcal_an

2583322445.7096844

In [54]:
dispo_cereales_destinee_alim_ani_g_prot_an = e['Aliments pour animaux en poids protéines/an'].sum()*1000
dispo_cereales_destinee_alim_ani_g_prot_an

66463820.40781978

In [55]:
f = (d.loc[(d['Nature Produit 3']=="viandes"),:])
f

Produit Nature Produit Nature Produit 2 Nature Produit 3  \
90  Viande d'Ovins/Caprins          autre           animal          viandes   
92        Viande de Bovins          autre           animal          viandes   
93        Viande de Suides          autre           animal          viandes   
94     Viande de Volailles          autre           animal          viandes   
95           Viande, Autre          autre           animal          viandes   

    Aliments pour animaux (en milliers de tonnes)  \
90                                           18.0   
92                                            9.0   
93                                            0.0   
94                                            0.0   
95                                           57.0   

    Nourriture (en milliers de tonnes)  Disponibilité alimentaire (Kcal/an)  \
90                             13389.0                         2.756593e+10   
92                             65216.0                         1.011677e+11   
93                            112101.0                         3.178415e+11   
94                            104870.0                         1.500870e+11   
95                              6823.0                         8.037495e+09   

    Disponibilité de protéines en quantité (Kg/an)  \
90                                    1.847171e+06   
92                                    9.031706e+06   
93                                    1.186332e+07   
94                                    1.317468e+07   
95                                    1.211730e+06   

    Disponibilité alimentaire en quantité (kg/an)  \
90                                   1.338991e+07   
92                                   6.522316e+07   
93                                   1.120917e+08   
94                                   1.048793e+08   
95                                   6.828896e+06   

    ratio_poids_proteines_poids_total_en_kg_an  \
90                                    0.137952   
92                                    0.138474   
93                                    0.105836   
94                                    0.125618   
95                                    0.177442   

    ratio_energie_poids en Kcal/an  \
90                     2058.708725   
92                     1551.100604   
93                     2835.548121   
94                     1431.044854   
95                     1176.983070   

    Aliments pour animaux en poids protéines/an  \
90                                     2.483143   
92                                     1.246265   
93                                     0.000000   
94                                     0.000000   
95                                    10.114172   

    Aliments pour animaux en Kcal/an  
90                      37056.757050  
92                      13959.905435  
93                          0.000000  
94                          0.000000  
95                      67088.035006

In [56]:
dispo_animale_destinee_alim_humaine_kg_an = f['Nourriture (en milliers de tonnes)'].sum()
dispo_animale_destinee_alim_humaine_kg_an

302399.0

In [57]:
dispo_animale_destinee_alim_humaine_kcal_an = f['Disponibilité alimentaire (Kcal/an)'].sum()
dispo_animale_destinee_alim_humaine_kcal_an

604699588365.0

In [58]:
dispo_animale_destinee_alim_humaine_g_prot_an = f['Disponibilité de protéines en quantité (Kg/an)'].sum()
dispo_animale_destinee_alim_humaine_g_prot_an

37128610.84019999

In [59]:
ratio_kg_an = dispo_cereales_destinee_alim_ani_kg_an / dispo_animale_destinee_alim_humaine_kg_an
ratio_kg_an

2.888686801212967

In [60]:
ratio_kcal_an = dispo_cereales_destinee_alim_ani_kcal_an /dispo_animale_destinee_alim_humaine_kcal_an
ratio_kcal_an

0.004272075747057358

In [61]:
ratio_g_prot_an = dispo_cereales_destinee_alim_ani_g_prot_an /dispo_animale_destinee_alim_humaine_g_prot_an
ratio_g_prot_an

1.7900971489043132

Question : Combien de kg de céréales sont nécessaires pour produire 1kg de viande ?

In [62]:
kg_cereales_pour_1kg_viande = 1*ratio_kg_an
kg_cereales_pour_1kg_viande

2.888686801212967